pkgs/by-name/yo/your_spotify/client.nix

```diff
@@ -0,0 +1,54 @@
+{
+  apiEndpoint ? "localhost:8080",
+  fetchYarnDeps,
+  lib,
+  makeWrapper,
+  mkYarnPackage,
+  prefetch-yarn-deps,
+  src,
+  version,
+  yarn,
+}:
+mkYarnPackage rec {
+  inherit version src;
+  name = "your_spotify_client-${version}";
+  offlineCache = fetchYarnDeps {
+    yarnLock = src + "/yarn.lock";
+    hash = "sha256-pj6owoEPx9gdtFvXF8E89A+Thhe/7m0+OJU6Ttc6ooA=";
+  };
+  configurePhase = ''
+    runHook preConfigure
+
+    export HOME=$(mktemp -d)
+    yarn config --offline set yarn-offline-mirror $offlineCache
+    fixup-yarn-lock yarn.lock
+    yarn install --offline --frozen-lockfile --ignore-platform --ignore-scripts --no-progress --non-interactive
+    patchShebangs node_modules/
+
+    runHook postConfigure
+  '';
+  buildPhase = ''
+    runHook preBuild
+    pushd ./apps/client/
+    pwd
+    yarn --offline run build
+    popd
+    runHook postBuild
+  '';
+  nativeBuildInputs = [makeWrapper yarn prefetch-yarn-deps];
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r ./apps/client/build/* $out
+    substituteInPlace $out/variables-template.js --replace-quiet '__API_ENDPOINT__' "${apiEndpoint}"
+    mv $out/variables-template.js $out/variables.js
+  '';
+  doDist = false;
+  meta = with lib; {
+    homepage = "https://github.com/Yooooomi/your_spotify";
+    changelog = "https://github.com/Yooooomi/your_spotify/releases/tag/${version}";
+    description = "YourSpotify is a self-hosted application that tracks what you listen and offers you a dashboard to explore statistics about it!";
```

Comment:
We can inherit this from the main package

Reply:
Done